### PR TITLE
Throw when initialize has not been called

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:

- Replaces `BrowserUtils.blockNativeBrokerCalledBeforeInitialize` with `BrowserUtils.blockCallsBeforeInitialize` since now `initialize` call is required regardless of whether `allowNativeBroker` is enabled
- Calls `BrowserUtils.blockCallsBeforeInitialize` in all public APIs related to token acquisition and accounts
- Replaces `BrowserAuthError.createNativeBrokerCalledBeforeInitialize()` with `BrowserAuthError.createUninitializedPublicClientApplication()` and updates error message
- Updates browser spec files to match new behavior

### CODE REVIEW

File: lib/msal-browser/src/controllers/StandardController.ts
Comment (tnorling) [inline lib/msal-browser/src/controllers/StandardController.ts]: `getNativeAccountId` shouldn't be public so this isn't needed here

**Review by tnorling — APPROVED** [review]
Looks good besides one minor comment, thanks for this!

File: lib/msal-browser/src/controllers/StandardController.ts
Comment (hectormmg) [reply lib/msal-browser/src/controllers/StandardController.ts]: Are we planning on changing this method signature? Or are we waiting for v4 to not break the surface API?

Comment (tnorling) [reply lib/msal-browser/src/controllers/StandardController.ts]: We should change it in v3, it's not public surface - I'm fairly certain this got changed during the move to the StandardController. If it's needed in 1P let's make it protected

**Review by sameerag — COMMENTED** [review]
Can we not initialize ourselves by default instead of adding this check at a zillion places?

**Review by sameerag — CHANGES_REQUESTED** [review]
Can we sync up once before I approve this on why we chose not to initialize ourselves or if there is a better way to do this?

**Comment by hectormmg** [comment]
> Can we sync up once before I approve this on why we chose not to initialize ourselves or if there is a better way to do this?

There's not much we can do since initialize is async. We need these checks because someone could instantiate a PCA and start calling APIs without initializing.

**Review by sameerag — APPROVED** [review]
As discussed, approving with suggestions to improve on this for the next major version. Also:
nit: Can we rename the function name to `blockAPICallsBeforeInitialize()`?

**Review by shoatman — APPROVED** [review]
lgtm

**Review by tnorling — APPROVED** [review]

**Pull request merged**
